lib/src/yaml_edit/src/strings.dart

```diff
@@ -0,0 +1,313 @@
+  /// `\r\n` already
+  return result.replaceAll('\n', lineEnding + ' ' * indentation);
+}
+
+/// Returns [value] with the necessary formatting applied in a flow context
+/// if possible.
+///
+/// If [value] is a [YamlScalar], we try to respect its [style] parameter where
+/// possible. Certain cases make this impossible (e.g. a plain string scalar that
+/// starts with '>'), in which case we will produce [value] with default styling
+/// options.
+String getFlowScalar(Object value) {
+  if (value is YamlScalar) {
+    assertValidScalar(value.value);
+
+    if (value.value is String) {
+      if (hasUnprintableCharacters(value.value) ||
+          value.style == ScalarStyle.DOUBLE_QUOTED) {
+        return _getDoubleQuotedString(value.value);
+      }
+
+      if (value.style == ScalarStyle.SINGLE_QUOTED) {
+        return _getSingleQuotedString(value.value);
+      }
+    }
+
+    return _tryGetPlainString(value.value);
+  }
+
+  assertValidScalar(value);
+  return _tryGetPlainString(value);
+}
+
+/// Returns [value] with the necessary formatting applied in a block context
+/// if possible.
+///
+/// If [value] is a [YamlScalar], we try to respect its [style] parameter where
+/// possible. Certain cases make this impossible (e.g. a folded string scalar
+/// 'null'), in which case we will produce [value] with default styling
+/// options.
+String getBlockScalar(Object value, int indentation, String lineEnding) {
+  ArgumentError.checkNotNull(indentation, 'indentation');
+  ArgumentError.checkNotNull(lineEnding, 'lineEnding');
+
+  if (value is YamlScalar) {
+    assertValidScalar(value.value);
+
+    if (value.value is String) {
+      if (hasUnprintableCharacters(value.value)) {
+        return _getDoubleQuotedString(value.value);
+      }
+
+      if (value.style == ScalarStyle.SINGLE_QUOTED) {
+        return _getSingleQuotedString(value.value);
+      }
+
+      // Strings with only white spaces will cause a misparsing
+      if (value.value.trim().length == value.value.length &&
+          value.value.length != 0) {
+        if (value.style == ScalarStyle.FOLDED) {
+          return _getFoldedString(value.value, indentation, lineEnding);
+        }
+
+        if (value.style == ScalarStyle.LITERAL) {
+          return _getLiteralString(value.value, indentation, lineEnding);
+        }
+      }
+    }
+
+    return _tryGetPlainString(value.value);
+  }
+
+  assertValidScalar(value);
+
+  /// The remainder of the possibilities are similar to how [getFlowScalar]
+  /// treats [value].
+  return getFlowScalar(value);
+}
+
+/// Returns [value] with the necessary formatting applied in a flow context.
+///
+/// If [value] is a [YamlNode], we try to respect its [style] parameter where
+/// possible. Certain cases make this impossible (e.g. a plain string scalar
+/// that starts with '>', a child having a block style parameters), in which
+/// case we will produce [value] with default styling options.
+String getFlowString(Object value) {
```

Comment:
ah I didn't use to wrap before passing it in - fixed it though!

So with `YamlMap.wrap` we can do something like:

```
  final doc = YamlMap.wrap({'a': File('hi.dart')});
  print(doc['a'].existsSync());
```

I'll modify `wrapAsYamlNode` to check even `YamlNodes`

Reply:
Hehe, I guess it's fine for `wrapAsYamlNode` to throw an `ArgumentError` if the value (or child of said value) passed in can't be wrapped as an `YamlNode`..

I guess it's also easier to get this error sooner rather than later..